### DISCUSSION TITLE
pika.exceptions.StreamLostError: Transport indicated EOF

### DISCUSSION THREAD

**Discussion by sermonlizhi** [pika/pika#1451] [Q&A]
Hi:
  
I'm using NGINX to proxy RabbitMQ, and I can successfully send messages. However, when I use `start_consuming()`, I consistently encounter the error 'pika.exceptions.StreamLostError: Transport indicated EOF.' In the same environment, I can consume messages using the RabbitMQ Java client.

**Comment by lukebakken** [comment]
Thanks for using Pika and RabbitMQ.

You are probably blocking Pika's I/O loop. Pika has no notion of threading. Your message handler is probably taking a very long time to process messages, and Pika can't send or receive heartbeats.

Since you didn't provide code, I can't assist further.

**Comment by sermonlizhi** [comment]
```python
import traceback
import pika
from loguru import logger

AZURE_MQ_PROSY_HOST = 'rabbitmq.aa.cn'
AZURE_MQ_PORT = 5672
AZURE_MQ_USERNAME = 'rabbitmq'
AZURE_MQ_PASSWORD = 'rabbitmq'
def consume():
    queue_name = 'testEn.231024_161147_JP2.15_heartbeat_queue'
    exchange_name = 'heartbeat_job_post'
    topic = 'testEn.231024_161147_JP2.15'


    credentials = pika.PlainCredentials(AZURE_MQ_USERNAME, AZURE_MQ_PASSWORD)
    parameters = pika.ConnectionParameters(host=AZURE_MQ_PROSY_HOST,
                                           port=AZURE_MQ_PORT,
                                           credentials=credentials,
                                           heartbeat=0,
                                           socket_timeout=7)

    connection = pika.BlockingConnection(parameters)
    channel = connection.channel()
    channel.queue_declare(queue=queue_name, auto_delete=False)
    channel.exchange_declare(exchange=exchange_name, exchange_type='topic')
    channel.queue_bind(queue=queue_name, exchange=exchange_name, routing_key=topic)
    try:
        channel.basic_consume(queue=queue_name, on_message_callback=on_message_heartbeat_callback, auto_ack=False)
        channel.start_consuming()
    except:
        logger.error(traceback.format_exc())

def on_message_heartbeat_callback(channel, method, properties, body):
    msg = None
    try:
        msg = body.decode('gbk')
        logger.info(f'on_message_callback msg is{msg}')
    except:
        logger.error(traceback.format_exc())
    if msg is None:
        try:
            msg = body.decode('utf-8')
        except:
            logger.error(traceback.format_exc())
    # 新收到消息
    logger.info(" [x] Received %r" % msg)
    channel.basic_ack(delivery_tag=method.delivery_tag)  # ack机制


if __name__ == '__main__':
    consume()
```
the code above specifies， rabbitmq.aa.cn as the nginx service, and nginx proxies the TCP requests to the actual rabbitmq server

**Reply by lukebakken** [reply]
Questions -

* This code, as is, causes the error? You're not doing anything with the messages in your production environment?
* What do RabbitMQ and Nginx log at the same time as the error?
* What version of this library are you using?
* If you do not use ngnix, do you see this error?
* Please **attach** your full nginx configuration to this discussion. Do not paste it, attach it.

**Comment by BorjaArroyo** [comment]
Hi,

I would like to comment further on this. I am using RabbitMQ to establish a manager-worker pattern in which tasks are delivered from the manager to the workers. The manager is a FastAPI server and eventually gets disconnected. It is true that each task lasts a long time, but shouldn't this affect to the worker instead of the manager?

Thank you in advance.
